### ISSUE TITLE
Program not opening. 

### ISSUE THREAD

**Issue by KartMakerBrosU** [RetroAndDev/WiiUCafeSDKDeluxe#3]
Ok, so I'm trying to find this "Build" folder that your tutorial mentions, but I cant seem to find it. I've followed everything so far and I've even built my game using these unity settings ( hopefully they're correct): 

Build Debug level:         Development [ No Network]
Build Output:                 Unpackaged
Boot Mode:                    PCFS
Autoconnect Profiler:     Unchecked
Script Debugging:          Unchecked
Scripts Only Build:          Unchecked

After I build, I place your programs folder next to the folder of the build, but whenever I try to run your program, nothing comes up. I am stuck here to any help would be greatly appreciated.

**Comment by RetroAndDev** [comment]
Hi! 
First you build settings are wrong.
You need to build as
Build Output : Download Image
Boot Mode : NAND

Before building you need to setup Wii U Cafe SDK Deluxe. If you can't open it, try to reinstall Net Framework 4.8 runtime from [here](https://dotnet.microsoft.com/en-us/download/dotnet-framework/thank-you/net48-offline-installer)

So before making your build : 
#1 : Inside your Unity build folder, It can be where you want, but I recommend you to place it at <UnityProject>/Builds
#2 : Download and extract to your <UnityProject>/Builds/ folder the content of Wii U Cafe SDK Deluxe. You can download it from the release section. (You copy the folder inside the ZIP to the <UnityProject>/Builds/ folder to be precise)
#3 : Open WiiUCafeSDKDeluxe.exe, and configure your project.
#4 : Create a new version inside Wii U Cafe SDK Deluxe, name it as you want.
#5 : Make your Build in Unity.
To build you will target Wii U (logic) as "Download Image" (Build Output) and "NAND" (Boot Mode). You will build inside the version folder (the one you created and not "Versions" who contains tool's versions information)
Example : 
If you created a version "MyVersion", you will select the target build folder like <UnityProject>/Builds/MyVersion. He "normally" be created by tool when you created a version.
#6 : Relaunch Wii U Cafe SDK Deluxe and the section about your version will be filled. The size of "copyToSdCard" will be filled with the app size. Now you can use Wii U Cafe SDK Deluxe !

I will make an installer and a new installing tutorial in next months (when the last part of the "How to make a Wii U Game With Unity" will release on GBAtemp)
Do that help ?

**Comment by KartMakerBrosU** [comment]
My computer wont let me install .NET Framework 4.8 since it says I have 4.8.1, so I'm confused as to how that wont work. Still cant open the program, but as least you are making a new installer and tutorial, so I guess I'll just wait for that. 
The unity build that comes out has a copyToSDCard folder in it with .app files. I think I can use this with WUP Installer GX2 to add it to my Wii U, right?

**Comment by RetroAndDev** [comment]
> I think I can use this with WUP Installer GX2 to add it to my Wii U, right?
It will not let you install them cause of encryption. What why Wii U Cafe SDK Exists. It decrypt, unpack, repack and re-encrypt these for Retail Wii U
That is strange if you can't run/launch the tool, I think you use the latest version ?
Do you have a message or something to guide me ? 
Try to uninstall and reinstall 4.8.1. Your install can be corrupted

**Comment by KartMakerBrosU** [comment]
Yeah, my install might be corrupted, its showing up in some places and not others (where it also should be) I do not know how to fix this. I've looked for help but their options aren't helping. My best guess is that ill have to do some windows reinstall while keeping all of my files.

**Comment by RetroAndDev** [comment]
Keep me informed !

**Comment by KartMakerBrosU** [comment]
yeah, no, i tried to run in on my Windows 11 laptop that also claims to have a .net framework of 4.8 or later, and it still doesn't open.

**Comment by KartMakerBrosU** [comment]
Ok, im attempting to run the program from source, and it seems to be trying to find a file called "settings.json" from my roaming appdata. Is the lack of this file causing this program to not be running?

**Comment by KartMakerBrosU** [comment]
Ok, commenting out the need to call the json file got the program to boot.

**Comment by RetroAndDev** [comment]
It says that the file is not found ? Maybe I don't write the default settings to the settings.json file. I will check that

**Comment by KartMakerBrosU** [comment]
Ok, I was able to (somehow) get the settings.json file created. The program opens and works fine. However, when I click "Make Master for my Hardware", it creates the folder, but there is nothing inside of it.  I make sure that there are no errors in App Config Tool, and I tried for Master Editor, but there are some in Basic Information and Advanced Configuration Information that I cannot edit. Those being Application Name and Max Memory size. I'm looking at the instructions you provided and the GBATemp comments and nobody seems to have this issue, so I'm just a little stumped here.

**Comment by RetroAndDev** [comment]
Okay, and one time more I think my tutorial is very bad (bad original French instructions and translated to my English...). The installation setup is how to say that... specific. 
So,
First of all, I will use C:/Users/RetroAndDev/UnityWiiUProject/ as Unity Project main path. (parent folder of Assets, Library...)
Inside it create a new folder "Builds". Download from release page the latest version of WiiUCafeSDKDeluxe. You will copy the folder inside the downloaded zip "WiiUCafeSDKDeluxe-vX.X.X" to <UnityProject>/Builds/. The full path to WiiUCafeSDKDeluxe.exe will be C:/Users/RetroAndDev/UnityWiiUProject/Builds/WiiUCafeSDKDeluxe-vX.X.X/WiiUCafeSDKDeluxe.exe.
You can now start the tool, setup projet and create the first version (inside WiiUCafeSDK Deluxe, like in the GBATemp tutorial). For now, I will use a version "VersionA" as test.
When the version "VersionA" is created you will normaly see a folder "VersionA" at <UnityProject>/Builds/WiiUCafeSDKDeluxe-VX.X.X/VersionA
You will build now the Unity Project inside this "VersionA" Folder with these settings : 
![WiiUBuildSettings](https://github.com/RetroAndDev/WiiUCafeSDKDeluxe/assets/56597499/96fcac04-2e80-40f3-aa14-4f89b768e2b5)
After the Unity build, check if output is like : <UnityProject>/Builds/VersionA/copyToSDCard/<a lot of files>
If you have this, you can now use WiiU Cafe SDK Deluxe to master the Wii U Dev files from "copyToSDCard" for a Retail Wii U Unit

**Comment by KartMakerBrosU** [comment]
Alright, cool. I did all that and now it successfully installs on my Wii U. Now it says I cant play it because I never bought it from the Wii U E-shop. My guess is that Its not signed correctly or something. I don't think this is your fault though, you've done an amazing job putting up with my various questions about this tool, and I hope that V2 fixes a lot of the issues I had leading up to now, so thank you. By the way, if you are interested, I can help with proof reading and perhaps revisions of the  English version of the instructions if you ever need it.  Ive been using Unity and modding the Wii U for the past 6 years, so I have a pretty good grasp on a lot of the terminology and concepts that this project uses to create more succinct instructions. You dont have to, its okay, have a wonderful day though, and again, thanks for all of your help.

**Comment by RetroAndDev** [comment]
I will help you to finish your project.
I make this tool and I made some shit on some points

So, with your build, you install with what ? System Config Tool ? or with other thing(s) ?
If the Wii U says that you don't have the game on e-shop, check your titleID. That will be a new feature of the V2 with automatic titleID generation. You maybe have a game or tool titleID. Check here : http://wiiubrew.org/wiki/Title_database

Do you use my "How to make a Wii U Game with Unity" on GBAtemp ? I let the link : https://gbatemp.net/threads/make-a-unity-game-for-wii-u.633391/
You have the archive in the tutorial with SCT (System Config Tool).
I don't tell in the tutorial how to install Unity game with SCT (will be in the next upcoming part). I can tell you if you want.
Let me know !

**Comment by KartMakerBrosU** [comment]
1. I tried using  WUP Installer GX2 and NUSspli for installing the build.
2. I checked if my title ID was unique and it was, so I don't know if its that (It might be, idk)
3. Yeah, how would I install the game using SCT?

**Comment by RetroAndDev** [comment]
I tell you tomorrow (for me). I live in France, and I go to bed.

**Comment by RetroAndDev** [comment]
And I think that. Can you send me the build you installed on your Wii U ? I can test with SCT and tell if it that

**Comment by RetroAndDev** [comment]
Installed works BUT the game never loads.
It stuck on a black screen with loading music in background loop.
What version of Unity and WiiU Build support ? Do you completely follow my installation tutorial ?
Did you build was make with 
Build Debug Level : Development [No network]
Build Output : Download Image
Boot Mode : NAND
if not retry with these (also try with Master [No network] perhaps ?)

To use SCT, download it and install it from the Part One files from my GBAtemp tutorial (install it with WUP Installer)
NOTE : BE CAREFUL WITH SYSTEM CONFIG TOOL. IT CAN DESTROY YOUR WII U. I WILL DON'T BE RESPONSIBLE FOR ANY BRICKS
Copy to and folder of your SD your proceed Unity build from WiiUCafeSDKDeluxe and launch SCT.
Inside go to 
Data Manager > Title Manager > Install > SD Card > [go to your installable files] > Select it and press R trigger to install it. 
Go back home and exit SCT, you can run your game
tell me what worked for you

**Comment by KartMakerBrosU** [comment]
Im using Unity 2017.1.2p3 and yeah, I followed your tutorial up until the unity export. After that I followed your instructions on here and others found in both GBATemp posts. I trued exporting as Master [No Network] and Development [No Network] and I still get the whole "This software cannot be used. The NNID used to purchase the software has been deleted" thing. I tried installing with Sys Confic Tool and I get the same error.

**Comment by RetroAndDev** [comment]
Hummm. Strange, your Wii U is an USA version ? Last software updates ? Hacked with what ?
If there is something it due to the hack. I don't have much time to investigate why, but you can ask why on GBAtemp in Wii U Hack. Maybe there is something you have on your Nintendo Account that EUROPEAN accounts don't have.
But, when I will have more time investigate why the game won't load. Obtaining the logs will be much complicated that real dev kit but I will try

**Comment by KartMakerBrosU** [comment]
Yeah my Wii U is a USA Deluxe Model with the latest update on version 5.5.6. This might be an issue because the update only released in the US. But then again, I did install a Wii U version of Portal 2 into my Wii U and it opened fine so It might not be my console? This is really weird, do you have a test unity build that I can try to install onto my Wii U to see if it works?

**Comment by RetroAndDev** [comment]
Sure ! 
I send it to you by mail

**Comment by KartMakerBrosU** [comment]
Ok, good news! it does install and does load, so there's something up with my Unity project, or how i'm using WiiiUCSDKD.

**Comment by RetroAndDev** [comment]
Oh ! Perfect. Can you send me you Unity project (by mail), I can try this weekend to build it myself and install it. I can tell you what make the build don't install/launch

**Comment by KartMakerBrosU** [comment]
hello, did you get the email?

**Comment by RetroAndDev** [comment]
I check and I don't get something.

**Comment by RetroAndDev** [comment]
Yeah but no. When I talk about the Unity Project I talk about the sources to compile it myself

**Comment by KartMakerBrosU** [comment]
OH, 🤦 whoops. Uploaded the wrong file. Let me fix that up

**Comment by KartMakerBrosU** [comment]
Heres the correct link: I
https://drive.google.com/file/d/18IEnEn4exy0sblP-M9G4QGvgzkcVDxaV/view?usp=sharing
I knew what you were referring do, I just uploaded the wrong file. this is the correct one.

**Comment by KartMakerBrosU** [comment]
Were you able to find any issues?

**Comment by RetroAndDev** [comment]
I don't had the time to check the last few weeks. I'm in holidays next Saturday and I will check this day

**Comment by RetroAndDev** [comment]
Hey, The tutorial is done, it will release on 5 January 2024 with the last revision of Wii U Cafe SDK Deluxe (v1)

I will verify this week all the tutorial and your project. I tell you the reasons why it don't work quickly !

**Comment by RetroAndDev** [comment]
Closing as the V2 solve this issue
Please download the latest beta of the V2 from [releases](https://github.com/RetroAndDev/WiiUCafeSDKDeluxe/releases/tag/v2.0.0-beta1)
